zeroconf/__init__.py

```diff
@@ -719,6 +719,20 @@ def __init__(self, data: bytes) -> None:
         except (IndexError, struct.error, IncomingDecodeError):
             self.log_exception_warning(('Choked at offset %d while unpacking %r', self.offset, data))
 
+    def __repr__(self) -> str:
+        return '<DNSIncoming:{%s}>' % ', '.join(
+            [
+                'id=%s' % self.id,
+                'flags=%s' % self.flags,
+                'n_q=%s' % self.num_questions,
+                'n_ans=%s' % self.num_answers,
+                'n_auth=%s' % self.num_authorities,
+                'n_add=%s' % self.num_additionals,
+                'questions=%s' % self.questions,
+                'answers=%s' % self.answers,
+            ]
```

Comment:
I'd go for something like `join('%s=%s' % (name, getattr(self, name)) for name in ['id', 'flags', 'num_questions', ...])` to avoid repetition, but that'll do too.

Reply:
Also it'll be cool to use dataclasses once Python 3.5 compatibility is dropped (EOL 2020-09-13 but I think we can drop it earlier) so we get nice reprs for free.